scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1908,6 +1909,15 @@ class Router(formatOps: FormatOps) {
             Split(Newline, 1).withIndent(2, lastToken, After)
           )
         }
+      // type bounds in type definitions and declarations
```

Comment:
how do we guarantee that this rule only applies for scala3? (otherwise, it would change formatting for existing code)

Reply:
Currently, we might change the formatting for existing code, though Scala 2 only allows `<:` for declarations. We could check if the dialect is Scala3, which would make sure that we don't change this behaviour for existing users.

Reply:
@olafurpg thoughts on backwards compatibility here?
@tgodzik should we use `newlines.beforeMultiline` to make a decision here? (and for default, aka `classic`, case, i would leave formatting as was before.)

Reply:
I think it would be good to limit Scala 3 changes to Scala 3 users where possible. It’s too early IMO to introduce changes in Scala 2 formatting for hypothetical Scala 3 use-cases.

Reply:
I will check the dialect in this case then. As for `newlines.beforeMultiline` I don't think it really applies here. Usually it's after `=` or `=>`

Reply:
then `newlines.source`?

Reply:
Ok, so when we add that it means we would split on `&` and `|` instead of `<:` always besides `unfold`, I added instead the check for the Scala 3 dialect, which makes sure that we don't modify the behaviour for existing users.